projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -624,9 +624,15 @@ export class IgxOverlayService implements OnDestroy {
                 if (!info.elementRef.nativeElement.contains(ev.target)) {
                     // if we should exclude position target check if the click is over it. If so do not close overlay
                     const positionTarget = info.settings.positionStrategy.settings.target as HTMLElement;
-                    const clickOnPositionTarget = positionTarget && positionTarget.contains
-                        ? (positionTarget as any).contains(ev.target)
-                        : false;
+                    let clickOnPositionTarget = false;
+                    if (positionTarget) {
+                        const positionTargetRect = positionTarget.getBoundingClientRect();
+                        if (ev.clientX >= positionTargetRect.left && ev.clientX <= positionTargetRect.right &&
```

Comment:
You can assign directly clickOnPositionTarget like this:
```
clickOnPositionTarget = positionTargetRect.left <= ev.clientX  && ev.clientX <= positionTargetRect.right &&
                              positionTargetRect.top <= ev.clientY  && ev.clientY <= positionTargetRect.bottom;
```
You do not need the if statement.